### PR TITLE
Add geo location as ROS type

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref https://github.com/nextcloud/spreed/issues/1478

### CODE REVIEW

File: lib/public/RichObjectStrings/Definitions.php [hunk 1]
```diff
@@ -418,6 +418,37 @@ class Definitions {
...
+		'geo-location' => [
+			'author' => 'Nextcloud',
+			'app' => 'core',
+			'since' => '22.0.0',
+			'parameters' => [
+				'id' => [
+					'since' => '22.0.0',
+					'required' => true,
+					'description' => 'The geo link to identify the location',
+					'example' => 'geo:52.5450511,13.3741463',
```
Comment (PVince81) [inline lib/public/RichObjectStrings/Definitions.php:430]: assuming that the semantics of the id doesn't have to match whatever is in the attributes and can be anything ?

Comment (nickvergessen) [reply lib/public/RichObjectStrings/Definitions.php:430]: well we can also drop the attributes and force the clients to parse it from the id... I think parsing sucks and its better to have it explicit additional as an attribute. we can update the sentencens if you think that helps, ref https://en.wikipedia.org/wiki/Geo_URI_scheme

Comment (PVince81) [reply lib/public/RichObjectStrings/Definitions.php:430]: in general I'd say that keeping duplicate information in sync also sucks, but for this one use case the information will be set only once and never change, so I guess it's ok that way and there is low risk of discrepancies in the future

**Review by PVince81 — APPROVED** [review]
:+1:

**Review by danxuliu — CHANGES_REQUESTED** [review]
The _longitude_ and _latitude_ example values are inverted; `52.5450511` should be the latitude and `13.3741463` should be the longitude.

Also in the list of parameters it could be better to first describe _latitude_ and then _longitude_ so they match with the order in the geo URI scheme.

**Comment by nickvergessen** [comment]
Fixed @danxuliu

**Review by danxuliu — APPROVED** [review]

**Pull request merged**